src/core/mixer/audio/audio_mixer.cpp

```diff
@@ -100,9 +101,14 @@ struct audio_mixer::impl
 
         for (auto& item : items) {
             auto ptr  = item.samples.data();
-            auto size = std::min(item.samples.size(), result.size());
+            auto size = result.size();
             for (auto n = 0; n < size; ++n) {
-                mixed[n] = static_cast<double>(ptr[n]) * item.transform.volume + mixed[n];
+		if (n<item.samples.size()){
+                    mixed[n] = static_cast<double>(ptr[n]) * item.transform.volume + mixed[n];
+		}else{
+		    auto offset = (item.samples.size()) - (channels-(n%channels));
```

Comment:
What does this do?

Reply:
The core issue at hand is that the producers and mixer/consumers don't synchronize their use of the audio cadence. This code duplicates the last sample from each audio channel if the mixer is on a longer cadence value than the producer in question. In the opposite case, where the mixer wants a short cadence and the producer created a longer one, it simply drops the extra sample (or two, depending on the cadence). This method is undetectable in the real world, as duplicating the last value prevents a click or pop (as opposed to a zero value), and it never allows the audio to fall out of sync beyond a couple samples before self-correcting when the inverse case occurs. The max audio sync deviation is +/- ~1/800th of a frame.